.github/workflows/ci.yml

```diff
@@ -192,9 +192,8 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        # test recent mypy versions on 3.10 and mypy latest on all supported python versions
-        # (note pydantic V2 only supports mypy>1 and we don't yet support mypy 1.1.1)
-        mypy-version: ['1.0.1']
+        # test recent mypy versions on 3.10 and mypy 1.0.1 on all supported python versions
+        mypy-version: ['0.971', '0.982', '0.991', '1.0.1']
```

Comment:
we still want to remove all `mypy` before `1.0.1` - we're not going to support older mypy in V2.

Reply:
merge once this is done.

Reply:
Really? I thought we removed it primarily because of the slowdown. If we solve that with this config, wouldn’t we want to support older versions of mypy to avoid forcing users to migrate both packages at the same time? Upgrading mypy may produce a ton of new type errors in users code, I shiver at the thought of the huge PRs users would have to do to update to V2

Reply:
Humm, it depends how much extra code we require to keep old versions working.

I think the big problem will be mypy 1.1.1 and 1.2.

But you have a point about big updates to support newer mypy.